eis_toolkit/spatial_analyses/cba.py

```diff
@@ -0,0 +1,511 @@
+    is contained in the file name, it is used to indicate the coordinate
+    system (CRS).
+
+    Args:
+        input_csv_file_path: CSV filename.
+
+    Returns:
+        Corresponding CBA object.
+    """
+
+    tmp = input_csv_file_path.split("__")
+    crs = tmp[-1].split(".")[0].replace("_", ":")
+    df = pd.read_csv(input_csv_file_path)
+    df.insert(0, "geometry", df["geom"].apply(wkt.loads))
+    # df["geometry"] = df["geom"].apply(wkt.loads)
+    df = df.drop("geom", axis=1)
+    cba_grid = gpd.GeoDataFrame(df, geometry="geometry")
+    cba_grid = cba_grid.set_crs(crs)
+    cba_grid.index = df["cell_id"]
+    cba_grid = cba_grid.drop("cell_id", axis=1)
+    return cba_grid
+
+
+def from_vector_file(cba_vector_file_path: str) -> gpd.GeoDataFrame:  # type: ignore[no-any-unimported]
+    """Intermediate utility.
+
+    Read a CBA grid from a vector file (ESRI shapefile, geojson...).
+    All attributes (and all values of each attribute) are kept.
+    The geometries contained in the file are assumed to represent a regular
+    square grid.
+
+    Args:
+        cba_vector_file_path: input vector file.
+
+    Returns:
+        Corresponding CBA object.
+    """
+    geodata = gpd.GeoDataFrame.from_file(cba_vector_file_path)
+    cba_grid = gpd.GeoDataFrame(geodata, geometry="geometry")
+    cba_grid.set_index("cell_id", inplace=True)
+    cba_grid.crs = geodata.crs
+    cols = cba_grid.columns.tolist()
+    cols = cols[-1:] + cols[:-1]
+    cba_grid = cba_grid[cols]
+    return cba_grid
+
+
+def to_csv(cba: gpd.GeoDataFrame, output_path: str) -> None:
+    """Intermediate utility.
+
+    Saves the object in a CSV file.
+
+    Args:
+        output_csv_file_path: name of the saved file suffixed with the
+            CRS (spatial projection system).
+
+    Returns:
+        None
+    """
+
+    crs_txt = f"epsg:{cba.crs.to_epsg()}"
+    tmp = gpd.GeoDataFrame(cba)
+    tmp["geom"] = tmp.geometry.apply(lambda x: wkt.dumps(x))
+    tmp = tmp.drop("geometry", axis=1)
+    tmp["cell_id"] = cba.index
+    tmp.to_csv(output_path + "__" + str(crs_txt).replace(":", "_") + ".csv", index=False)
+
+
+def to_raster(cba: gpd.GeoDataFrame, output_path: str) -> None:
+    """Intermediate utility.
+
+    Saves the object as a raster TIFF file.
+
+    Args:
+        output_tiff_path: name of the saved file.
+
+    Returns:
+        None
+    """
+
+    crs_txt = f"EPSG:{cba.crs.to_epsg()}"
+    count = len(cba.columns.drop("geometry"))
+
+    geometries = cba["geometry"].values
+    x = np.unique(geometries.centroid.x)
+    y = np.unique(geometries.centroid.y)
+    X, Y = np.meshgrid(x, y)
+    x_resolution = X[0][1] - X[0][0]
+    y_resolution = Y[1][0] - Y[0][0]
+    min_x, min_y, max_x, max_y = cba.total_bounds
+    width = (max_x - min_x) / x_resolution
+    height = (max_y - min_y) / y_resolution
+
+    values = []
+    col_name = []
+    for col in cba.columns.drop("geometry"):
```

Comment:
Due to the use of `geom` column name, only dropping `geometry` is no sufficient as the `cba` `GeoDataFrame` can contain the `geom` column with `object` as dtype.

```suggestion
    for col in cba.columns.drop(labels=["geometry", "geom"]):
```

Alternatively, you can refactor out of using the `geom` column as suggested in another comment.

Reply:
Even with the `geom` column out of the algorithm, it would be good to avoid `geometry` type column in this function. This implementation can solve it I think.

```suggestion
    for col in cba.select_dtypes(exclude=['geometry']).columns:
```

Reply:
I believe this is still incorrect as `'geometry'` is not a defined dtype. See my new review below.